src/GafferScene/Instancer.cpp

```diff
@@ -272,8 +272,19 @@ int seedForPoint( int index, const PrimitiveVariable *primVar, int numSeeds, int
 	int id = index;
 	if( primVar )
 	{
-		// TODO - the exception this will throw on non-int primvars may not be very clear to users
-		id = PrimitiveVariable::IndexedView<int>( *primVar )[index];
+		if( primVar->data->typeId() == IntVectorData::staticTypeId() )
+		{
+			id = PrimitiveVariable::IndexedView<int>( *primVar )[index];
+		}
+		else if( primVar->data->typeId() == Int64VectorData::staticTypeId() )
+		{
+			id = PrimitiveVariable::IndexedView<int64_t>( *primVar )[index];
+		}
+		else
+		{
+			// We check the type of the primvar before we try to use it here
+			assert( false );
+		}
```

Comment:
How hard would it be to use an IdData here? Just wondering if the overhead of the type checking and IndexedView construction is measurable here.

Reply:
Done in [d5b7d42](https://github.com/GafferHQ/gaffer/pull/6088/commits/d5b7d4281f796cd5c3878ca862b2afb43b997ec9), and really wasn't that bad.

I had started down the path of pulling the seed out of m_prototypeContextVariables and handling it separately ... the actual data we need to store for it is pretty disjoint from the rest of the variables, but it ended up feeling easier to just keep the current structure.